### PR TITLE
[FLINK-22294][hive] Hive reading fail when getting file numbers on different filesystem nameservices

### PR DESCRIPTION
**Pull request by zuston**
…

## What is the purpose of the change

*(For example: This pull request makes task deployment go through the blob server, rather than through RPC. That way we avoid re-transferring them on each deployment (during recovery).)*

## Brief change log

*(for example:)*
  - *The TaskInfo is stored in the blob store on job creation time as a persistent artifact*
  - *Deployments RPC transmits only the blob storage reference*
  - *TaskManagers retrieve the TaskInfo from the blob cache*

## Verifying this change

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (100MB)*
  - *Extended integration test for recovery after master (JobManager) failure*
  - *Added test that validates that TaskInfo is transferred only once across recoveries*
  - *Manually verified the change by running a 4 node cluser with 2 JobManagers and 4 TaskManagers, a stateful streaming program, and killing one JobManager and two TaskManagers during the execution, verifying that recovery happens correctly.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no)
  - The serializers: (yes / no / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / no / don't know)
  - The S3 file system connector: (yes / no / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by zuston** [comment]
Could you review it @lirui-apache

File: flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/read/HiveTableInputFormat.java [hunk 1]
```diff
@@ -345,13 +345,10 @@ public InputSplitAssigner getInputSplitAssigner(HiveTableInputSplit[] inputSplit
 
     public int getNumFiles() throws IOException {
```
Comment (lirui-apache) [inline flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/read/HiveTableInputFormat.java:346]: It seems this method is no longer needed. We can just remove it.

Comment (zuston) [reply flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/read/HiveTableInputFormat.java:346]: Got it.

**Comment by lirui-apache** [comment]
@zuston Thanks for working on this issue. LGTM except for one minor comment.

**Comment by zuston** [comment]
> @zuston Thanks for working on this issue. LGTM except for one minor comment.

Thanks for you reply.

**Comment by zuston** [comment]
The test failure does not seem to be related to this PR, can you check it? @lirui-apache

**Review by lirui-apache — APPROVED** [review]
LGTM

**Comment by zuston** [comment]
Will merge? @lirui-apache

**Pull request closed without merging**